<?xml version="1.0"?>
<!-- This is a sample lanuch file, please change it based on your needs -->
<launch>
    <arg name="ip" default="192.168.1.1" />
    <arg name="freq" default="8" />
    <node name="ardrone_driver" pkg="ardrone_autonomy" type="ardrone_driver"
          output="screen" clear_params="true" args="-ip $(arg ip)">

        <!-- Modifies the drone's onboard parameters.
             If not specified, drone default will be used (consult SDK or ardrone_autonomy wiki) -->
        <param name="outdoor" value="1" />
        <param name="flight_without_shell" value="1" />
        <param name="navdata_demo" value="0" />

        <param name="max_bitrate" value="2000" />
        <param name="bitrate" value="2000" />
        <param name="video_codec" value="129" />

        <param name="altitude_max" value="5000" />
        <param name="altitude_min" value="300" />

        <param name="control_vz_max" value="2000" />
        <param name="control_yaw" value="6.11" />
        <param name="euler_angle_max" value="0.35" />

        <param name="detect_type" value="10" />
        <param name="detections_select_h" value="128" />
        <param name="detections_select_v_hsync" value="32" />
        <param name="enemy_colors" value="3" />
        <param name="enemy_without_shell" value="0" />
        <param name="ultrasound_freq" value="$(arg freq)" />

        <!-- Enables the standard /ardrone/navdata, imu and mag topics. If not specified, defaults to TRUE -->
        <param name="enable_legacy_navdata" value="true" />
        
        <!-- Enables the new-style, full information navdata packets. If not specified, defaults to FALSE -->
        <param name="enable_navdata_demo" value="true" />
        <param name="enable_navdata_time" value="true" />
        <param name="enable_navdata_raw_measures" value="true" />
        <param name="enable_navdata_phys_measures" value="true" />
        <param name="enable_navdata_gyros_offsets" value="true" />
        <param name="enable_navdata_euler_angles" value="true" />
        <param name="enable_navdata_references" value="true" />
        <param name="enable_navdata_trims" value="true" />
        <param name="enable_navdata_rc_references" value="true" />
        <param name="enable_navdata_pwm" value="true" />
        <param name="enable_navdata_altitude" value="true" />
        <param name="enable_navdata_vision_raw" value="true" />
        <param name="enable_navdata_vision_of" value="true" />
        <param name="enable_navdata_vision" value="true" />
        <param name="enable_navdata_vision_perf" value="true" />
        <param name="enable_navdata_trackers_send" value="true" />
        <param name="enable_navdata_vision_detect" value="true" />
        <param name="enable_navdata_watchdog" value="true" />
        <param name="enable_navdata_adc_data_frame" value="true" />
        <param name="enable_navdata_video_stream" value="true" />
        <param name="enable_navdata_games" value="true" />
        <param name="enable_navdata_pressure_raw" value="true" />
        <param name="enable_navdata_magneto" value="true" />
        <param name="enable_navdata_wind_speed" value="true" />
        <param name="enable_navdata_kalman_pressure" value="true" />
        <param name="enable_navdata_hdvideo_stream" value="true" />
        <param name="enable_navdata_wifi" value="true" />
        <param name="enable_navdata_zimmu_3000" value="true" />

        <!-- Tunes the speed at which the ros loop runs, and thus, the rate at which navdata is published -->
        <param name="looprate" value="50" />

        <!-- Do we want to publish new-style navdata when received (true), or every time the ros-loop runs (false)? -->
        <!-- (does not affect legacy navdata, which is always published at ros-loop rate) -->
        <param name="realtime_navdata" value="true" />
        <param name="realtime_video" value="true" />

        <!-- Covariance Values (3x3 matrices reshaped to 1x9)-->
        <rosparam param="cov/imu_la">[0.1, 0.0, 0.0, 0.0, 0.1, 0.0, 0.0, 0.0, 0.1]</rosparam>
        <rosparam param="cov/imu_av">[1.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 1.0]</rosparam>
        <rosparam param="cov/imu_or">[1.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 100000.0]</rosparam>
    </node>
</launch>
